c7n/resources/cognito.py

```diff
@@ -43,3 +81,37 @@ class resource_type(object):
         name = 'Name'
         filter_name = None
         dimension = None
+
+
+@CognitoUserPool.action_registry.register('delete')
+class DeleteUserPool(BaseAction):
+    """Action to delete cognito user pool
+
+    It is recommended to use a filter to avoid unwanted deletion of pools
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: user-pool-delete
+                resource: user-pool
+                actions:
+                  - delete
+    """
+
+    schema = type_schema('delete')
+    permissions = ("cognito-idp:DeleteUserPool",)
+
+    def process(self, pools):
+        with self.executor_factory(max_workers=2) as w:
+            list(w.map(self.process_pools, pools))
+
+    def process_pools(self, pools):
```

Comment:
thanks for the updates, the ones in this module looks like they also need an update.